rootex/framework/entity.cpp

```diff
@@ -66,13 +89,32 @@ bool Entity::setupComponents()
 bool Entity::setupEntities()
 {
 	bool status = true;
+
+	if (m_Script)
+	{
+		bool result = m_Script->setup();
+		if (!result)
+		{
+			m_Script.reset();
+		}
+	}
```

Comment:
This should ideally happen after the components have been set up. The script could possibly query the entity it is on for some components inside it at construction time (not necessarily the `onBegin()` function)